### PR TITLE
FEATURE: RAIL-1143 PivotTable: log warning into console when localIdentifiers are not unique

### PR DESCRIPTION
**Pull request by tomasreichmann**
…ntifiers are not unique

### CODE REVIEW

File: src/helpers/errorHandlers.ts [hunk 1]
```diff
@@ -137,3 +137,16 @@ export function checkEmptyResult(responses: Execution.IExecutionResponses) {
 
     return responses;
 }
+
+export const validateLocalIdentifiers = (buckets: VisualizationObject.IBucket[]) => {
```
Comment (lhorak) [inline src/helpers/errorHandlers.ts]: How about using uniq from lodash:

```
const localIdentifiers = buckets.map(bucket => bucket.localIdentifier);
const uniqueEntries = uniq(localIdentifiers);

return uniqueEntries.length === localIdentifiers.length;
```

Comment (lhorak) [inline src/helpers/errorHandlers.ts]: And I'd maybe rename it to hasDuplicateIdentifiers, as validate could mean more than finding duplicates

Comment (lhorak) [reply src/helpers/errorHandlers.ts]: Even better way, without map:

```
uniqBy(buckets, bucket => bucket.localIdentifier)
```

Comment (lhorak) [reply src/helpers/errorHandlers.ts]: maybe even 

```
uniqBy(buckets, 'localIdentifier');
```

is enough https://lodash.com/docs/4.17.10#uniqBy

Comment (lhorak) [reply src/helpers/errorHandlers.ts]: so oneliner :D 

```
const hasDuplicateIdentifiers = buckets => (buckets.length !== uniqBy(buckets, 'localIdentifier').length);
```

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1518/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1007/ : FAILURE in 35s

File: src/helpers/errorHandlers.ts (hunk 1 above)
Comment (tomasreichmann) [reply src/helpers/errorHandlers.ts]: done

Comment (tomasreichmann) [reply src/helpers/errorHandlers.ts]: solved differently

**Comment by lhorak** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/338/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1019/ : SUCCESS in 2m 57s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/596/ : SUCCESS in 26m 08s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/251/ : SUCCESS in 19m 53s

**Pull request merged**